tutor/src/models/scores/student.js

```diff
@@ -29,6 +29,15 @@ class Student extends BaseModel {
     return true !== this.is_dropped;
   }
 
+  get course() {
+    return this.period.course;
+  }
+
+  isValid(type) {
+    return this.course[`${type}_score_weight`] && !isNil(this[`${type}_score`]) &&
+      this.course[`${type}_progress_weight`] && !isNil(this[`${type}_progress`]);
+  }
+
```

Comment:
@brittany-johnson sorry for the large PR - most if it is just moving files around to match our standard layout, this is the actual bug.

Before we were just saying `!this.homework_score` here, however that'd also fail if `homework_score` was 0, when we really wanted to skip it only if it was null or undefined.  The `isNil` checks both of those, but not 0